src/frontend/tsx/GamePanel.tsx

```diff
@@ -22,46 +26,77 @@ export function GamePanel () {
+      console.log('CURRENT POSITION')
+      const currentPosition: ActiveGameInfo = {
+        position: displayGame.info.moves[displayGame.info.moves.length - 1].fen,
+        gotDrawOffer: displayGame.gotDrawOffer,
+        sentDrawOffer: displayGame.sentDrawOffer,
+        drawClaimAvailable: displayGame.drawClaimAvailable,
+        autoClaimSpecialDraws: displayGame.autoClaimSpecialDraws,
+        info: displayGame.info
+      }
+
+      setReviewMode(false)
+      setReviewIndex(null)
+      setDisplayGame(currentPosition)
+    }
+  }
+
+  const afterReviewIndex = (index: number) => {
+    if (reviewMode === true) {
+      if (index > reviewIndex) {
+        return true
+      } else {
+        return false
+      }
+    }
+  }
+
   return (
     <div className='game-panel-container col'>
       <div className="game-panel col">
-        <div id="opp-timer" className={'timer row' + (hasGame ? '' : ' invisible')}>
+        <div id="opp-timer" className={'timer row' + (hasGame ? '' : ' hidden')}>
           <p>00:00</p>
         </div>
-        <div id="opp-player" className={'player row' + (hasGame ? '' : ' invisible')}>
-          <p>~sampel-palnet</p>
+        <div id="opp-player" className={'player row' + (hasGame ? '' : ' hidden')}>
+          <p>{opponent}</p>
         </div>
-        <div className="moves col">
-          <div className="moves-divider"></div>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
+        <div className={'moves col' + (hasGame ? '' : ' hidden')}>
+          <ol>
+            {
+              Array.from(displayMoves).map((ply, thisIndex, thisArray) => {
+                const nextIndex: number = thisIndex + 1
+                if (thisIndex % 2 === 0) {
+                  return (
+                    <li key={ thisIndex } className='move-item' style={{ opacity: (afterReviewIndex(thisIndex) ? 0.4 : 1.0) }}>
+                      <span onClick={() => reviewPosition(thisIndex)}>{ply}</span>{ '\xa0'.repeat(6 - ply.length) }<span onClick={() => reviewPosition(nextIndex)} style={{ opacity: (afterReviewIndex(thisIndex) ? 1.0 : afterReviewIndex(nextIndex) ? 0.4 : 1.0) }}>{(nextIndex > thisArray.length ? '' : thisArray.at(nextIndex))}</span>
+                    </li>
+                  )
+                }
+              })
```

Comment:
There's a more verbose but more elegant way to do this:
```
for (const i: number = 0; i < displayMoves.length; i += 2) {
  const move: number = (i / 2) + 1
  const wMove: SAN = displayMoves.get(i)
  const bIndex: number = i + 1

  if (bIndex > displayMoves.length) {

    return (
      <li key={ move } className='move-item'>
        <span onClick={() => setDisplayIndex(wIndex)} style={{ opacity: moveOpacity(wIndex) }>
          { wMove }
        </span>
      </li>
    )
  } else {
    return (
      <li key={ move } className='move-item'>
        <span onClick={() => setDisplayIndex(wIndex)} style={{ opacity: moveOpacity(wIndex) }>
          { wMove }
        </span>
        //
        { '\xa0'.repeat(6 - wMove.length) }
        //
        <span onClick={() => setDisplayIndex(bIndex)} style={{ opacity: moveOpacity(bIndex) }>
          { displayMoves.get(bIndex) }
        </span>
      </li>
    )
  }
}
```

Note that the above code fragment:
- Renames `reviewIndex`/`setReviewIndex` to `displayIndex`/`setDisplayIndex`
- Removes `reviewPosition`
- Rewrites `displayMoves` as a local constant which pulls the SAN data out of the `Array` of FEN-SAN pairs for the displayed game
- Replaces `afterReviewIndex` with the below function, which applies at the ply level now:
```
const moveOpacity = (index: number) => {
  (index > reviewIndex) : 0.4 ? 1.0
}
```

See my next comment regarding my thoughts on `reviewMode`/`reviewIndex`.

Reply:
@bonbud-macryg Looks like `for` loops can't return anything, so my code block above would need to be edited slightly:
```
let components = [];
for (const i: number = 0; i < displayMoves.length; i += 2) {
  ...
  if (bIndex > displayMoves.length) {
    components.push(...)
  } else {
    components.push(...)
  }
}
return components
```

Reply:
This is kind-of working in the latest commit. I think the error is because it's enclosed in `Array.from` and it looks like the whole `for` loop should be replacing that. But as discussed, just putting it in curly brackets isn't accepted.